rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -1766,6 +2328,8 @@ rmw_destroy_service(rmw_node_t * node, rmw_service_t * service)
     return RMW_RET_ERROR;
   }
   rmw_service_free(service);
+
```

Comment:
Before calling `rmf_service_free()` we should deallocate everything in `service->data`. Including calling `z_drop()` on any `z_owned_XX` objects

Reply:
Done.